crates/ra_hir_ty/src/expr.rs

```diff
@@ -312,3 +386,132 @@ pub fn record_pattern_missing_fields(
     }
     Some((variant_def, missed_fields, exhaustive))
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
I suggest using `expect` crate for new test modules (and for new tests in general) :-)

Reply:
Specifically, the pattern that I find works best is defining and

```rust
fn check(ra_fixture: &str, expect: Expt) {
}

function in the test module, and then every test just calls this one. 
```